tests/unit_tests/gui/test_main_window.py

```diff
@@ -60,53 +64,134 @@ def opened_main_window(source_root, tmpdir_factory, request):
 
 @pytest.mark.usefixtures("use_tmpdir")
 @pytest.fixture(scope="module")
-def esmda_has_run(opened_main_window, request):
-    qtbot = QtBot(request)
-    gui = opened_main_window
-    qtbot.addWidget(gui)
-
+def esmda_has_run(opened_main_window, run_experiment):
     # Select Multiple Data Assimilation in the simulation panel
-    simulation_panel = gui.findChild(SimulationPanel)
-    assert isinstance(simulation_panel, SimulationPanel)
-    simulation_mode_combo = simulation_panel.findChild(QComboBox)
-    assert isinstance(simulation_mode_combo, QComboBox)
-    current_select = 0
-    simulation_mode_combo.setCurrentIndex(current_select)
-    while simulation_mode_combo.currentText() != MultipleDataAssimilation.name():
-        current_select += 1
+    run_experiment(MultipleDataAssimilation)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+@pytest.fixture(scope="module")
+def run_experiment(request, opened_main_window):
+    def func(experiment_mode):
+        qtbot = QtBot(request)
```

Comment:
The docs of `QtBot` say that:

    .. important:: Instances of this class should be accessed only by using a ``qtbot`` fixture,
                    never instantiated directly.
                    
Do we have a good reason for not following this recommended practice?

Reply:
No, not really, I just moved the existing code, did not think about it, it is created from `request`, _assume_  that must be relevant. But not really sure.

Reply:
Found the reason, it is scope mismatch:
```
ScopeMismatch: You tried to access the function scoped fixture qtbot with a module scoped request object, involved factories:
```